### PR TITLE
[compiler-rt][AArch64][FMV] Use the hw.optional.arm.caps fast path

### PR DESCRIPTION
**Pull request by jroelofs**
MacOS 15.0 and iOS 18.0 added a new sysctl to fetch a bitvector of all the hw.optional.arm.FEAT_*'s in one go.  Using this has a perf advantage over doing multiple round-trips to the kernel and back, but since it's not present in older oses, we still need the slow fallback.

### CODE REVIEW

File: compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc [hunk 1]
```diff
@@ -21,6 +26,63 @@ void __init_cpu_features_resolver(void) {
...
+    CHECK_BIT(CAP_BIT_FEAT_LRCPC2, FEAT_RCPC2);
+    CHECK_BIT(CAP_BIT_FEAT_FCMA, FEAT_FCMA);
+    CHECK_BIT(CAP_BIT_FEAT_JSCVT, FEAT_JSCVT);
+    CHECK_BIT(CAP_BIT_FEAT_DPB, FEAT_DPB);
+    CHECK_BIT(CAP_BIT_FEAT_DPB2, FEAT_DPB2);
+    CHECK_BIT(CAP_BIT_FEAT_BF16, FEAT_BF16);
+    CHECK_BIT(CAP_BIT_FEAT_I8MM, FEAT_I8MM);
+    CHECK_BIT(CAP_BIT_FEAT_DIT, FEAT_DIT);
+    CHECK_BIT(CAP_BIT_FEAT_FP16, FEAT_FP16);
+    CHECK_BIT(CAP_BIT_FEAT_SSBS, FEAT_SSBS2);
```
Comment (labrinea) [inline compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:68]: As I said on the other review SSBS2 implies SSBS.

**Review by labrinea — APPROVED** [review]
Looks good. I noticed MTE/2 are missing, perhaps other features too? Is that deliberate?

**Comment by saagarjha** [comment]
Note that `CAP_BIT_FEAT_SME`/`CAP_BIT_FEAT_SME2`/`CAP_BIT_FEAT_SME_F64F64`/`CAP_BIT_FEAT_SME_I16I64` are still stripped out of the public SDK, which breaks the build of compiler-rt :( I've filed FB13964283 to get those added but until then I'd also check for e.g. `CAP_BIT_FEAT_SME` before trying to use these.

**Comment by jroelofs** [comment]
Oops. Thanks. I'll put up another PR.

**Comment by jroelofs** [comment]
0c0281130ed51fea06cf20a2db37c5bca257ad31

**Pull request merged**